src/index.ts

```diff
@@ -70,22 +70,19 @@ type InitPreContextOut<T> = T extends HasInitPreContext<any, any>
   ? ReturnType<T['initPreContext']>
   : never;
 
```

Comment:
now sanitizeParams is different than the others...  I'd have preferred you did this in a separate issue/branch, but if you're really gonna do this in this issue, do it right:
1. change the code down there back to referencing this type (for consistency)
2. put these back, but change them to `type SanitizeParamsContextIn<T extends HasSanitizeParams<any, any>> = Parameters<T['sanitizeParams']>[0]`
3. Do the same to all the other *ContextIn and *ContextOut in this section (i.e. change them from being conditionals to simple types but with type constraints).
- [ ] bring back and make consistent

Reply:
- [ ] ^ likewise with the newly added stages

Reply:
done